int32 distance_input
